pkg/operator/kuryr.go

```diff
@@ -0,0 +1,146 @@
+// and some other small things.
+func (h *Handler) renderKuryr(c *v1.KuryrConfig) ([]*uns.Unstructured, error) {
+	operConfig := h.config.Spec
+	objs := []*uns.Unstructured{}
+
+	// generate master network configuration
+	ippools := []netv1.ClusterNetworkEntry{}
+	for _, net := range operConfig.ClusterNetworks {
+		ippools = append(ippools, netv1.ClusterNetworkEntry{CIDR: net.CIDR, HostSubnetLength: net.HostSubnetLength})
+	}
+
+	clusterNet := netv1.ClusterNetwork{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "network.openshift.io/v1",
+			Kind:       "ClusterNetwork",
+		},
+		ObjectMeta: metav1.ObjectMeta{Name: netv1.ClusterNetworkDefault},
+
+		ServiceNetwork:  operConfig.ServiceNetwork,
+		PluginName:      sdnPluginName("OpenStack Kuryr-Kubernetes"),
+		ClusterNetworks: ippools,
+		// VXLANPort:       c.VXLANPort,
+
+		Network:          ippools[0].CIDR,
+		HostSubnetLength: ippools[0].HostSubnetLength,
+	}
+	obj, err := k8sutil.UnstructuredFromRuntimeObject(&clusterNet)
+	if err != nil {
+		// This is very unlikely
+		return nil, errors.Wrap(err, "failed to transmutate ClusterNetwork")
+	}
+	objs = append(objs, obj)
+
+	// render the manifests on disk
+	data := render.MakeRenderData()
+
+	// OpenStack credentials
+	if c.OpenStackCACertificate == nil {
+		data.Data["OpenStackCACertificate"] = ""
+	} else {
+		data.Data["OpenStackCACertificate"] = *c.OpenStackCACertificate
+	}
+	data.Data["OpenStackAuthURL"] = *c.OpenStackCredentials.AuthURL
+	data.Data["OpenStackPassword"] = *c.OpenStackCredentials.Password
+	data.Data["OpenStackProjectDomainName"] = *c.OpenStackCredentials.ProjectDomainName
+	data.Data["OpenStackProjectId"] = *c.OpenStackCredentials.ProjectId
+	data.Data["OpenStackUserDomainName"] = *c.OpenStackCredentials.UserDomainName
+	data.Data["OpenStackUsername"] = *c.OpenStackCredentials.Username
+	data.Data["KuryrProject"] = c.KuryrProject
+
+	// TODO(dulek): Figure out how do I get those
+	data.Data["PodSubnet"] = ""
+	data.Data["PodSecurityGroups"] = ""
+	data.Data["ServiceSubnet"] = ""
+	data.Data["WorkerNodesSubnet"] = ""
+
+	// daemon DaemonSet data
+	if c.DaemonEnableProbes == nil {
+		data.Data["DaemonEnableProbes"] = "false"
+	} else {
+		data.Data["DaemonEnableProbes"] = *c.DaemonEnableProbes
+	}
+	if c.DaemonProbesPort == nil {
+		data.Data["DaemonProbesPort"] = 8090
+	} else {
+		data.Data["DaemonProbesPort"] = *c.DaemonProbesPort
+	}
+	if c.CNIBinDir == nil {
+		data.Data["CNIBinDir"] = "/opt/cni/bin"
+	} else {
+		data.Data["CNIBinDir"] = *c.CNIBinDir
+	}
+
+	// controller Deployment data
+	if c.ControllerEnableProbes == nil {
+		data.Data["ControllerEnableProbes"] = "true"
+	} else {
+		data.Data["ControllerEnableProbes"] = *c.ControllerEnableProbes
+	}
+	if c.ControllerProbesPort == nil {
+		data.Data["ControllerProbesPort"] = 8082
+	} else {
+		data.Data["ControllerProbesPort"] = *c.ControllerProbesPort
+	}
+
+	// TODO(dulek): How do I get image names correctly?
+	data.Data["OpenShiftSDNImage"] = ""
```

Comment:
Why do you need the openshiftSDNImage?

Reply:
To get the OpenShift-grade CNI binaries (mainly loopback) and then to be able to run the DNS container. See 005-daemon.yaml and initContainer and sky-dns container there.